lib/types.js

```diff
@@ -82,7 +87,7 @@ function registerType(name, options) {
  */
 function listBuilder(list, bufb, codec, width) {
   if (!Array.isArray(list)) {
-    throw new errors.EncodingError('Unsure how to encode non-array as list');
+    throw new errors.EncodingError(list, 'Unsure how to encode non-array as list');
```

Comment:
Good change, I like including the value in the error, why didn't I think of that?